pkg/router/base_rule.go

```diff
@@ -122,13 +127,41 @@ func NewRouteRuleImplBase(vHost *VirtualHostImpl, route *v2.Router) (*RouteRuleI
 			body:   route.DirectResponse.Body,
 		}
 	}
+	// add redirect repsonse rule
+	if route.Redirect != nil {
+		r := route.Redirect
+		rule := &redirectResponseImpl{
+			path:   r.PathRedirect,
+			host:   r.HostRedirect,
+			scheme: r.SchemeRedirect,
+		}
+		switch r.ResponseCode {
+		case 0:
+			// default to 301
+			rule.code = http.StatusMovedPermanently
+		case http.StatusMovedPermanently, http.StatusFound,
+			http.StatusSeeOther, http.StatusTemporaryRedirect,
+			http.StatusPermanentRedirect:
+			rule.code = r.ResponseCode
+		default:
+			return nil, fmt.Errorf("redirect code not supported yet: %d", r.ResponseCode)
+		}
+		base.redirectResponseRule = rule
+	}
 	return base, nil
 }
 
 func (rri *RouteRuleImplBase) DirectResponseRule() api.DirectResponseRule {
 	return rri.directResponseRule
 }
 
+func (rri *RouteRuleImplBase) RedirectResponseRule() api.RedirectResponseRule {
+	if rri.redirectResponseRule == nil {
```

Comment:
It looks like it's equivalent to just return `rri.redirectResponseRule` ?

Reply:
actually they are subtly different. if `rri.redirectResponseRule` is nil and we simply return `rri.redirectResponseRule` here, `RedirectResponseRule() == (*redirectResponseImpl)(nil)`  rather than `nil`, and we have to call `reflect.ValueOf(..).IsNil()` at https://github.com/mosn/mosn/pull/1261/files#diff-ba11a00bb14720d45706e6dee657a6aeR713 to check if it is actually `nil`.

Reply:
you could check the example here https://play.golang.org/p/y-ZgiSzIo1r

Reply:
Cool, you are right.